src/node_dotenv.cc

```diff
@@ -98,73 +110,46 @@ bool Dotenv::ParsePath(const std::string_view path) {
     result.append(buf.base, r);
   }
 
-  using std::string_view_literals::operator""sv;
-  auto lines = SplitString(result, "\n"sv);
+  // Convert line breaks to the same format
+  std::string lines = result;
+  std::regex_replace(lines, std::regex("\r\n?"), "\n");
 
-  for (const auto& line : lines) {
-    ParseLine(line);
-  }
-  return true;
-}
+  std::smatch match;
+  while (std::regex_search(lines, match, LINE)) {
+    const std::string key = match[1].str();
 
-void Dotenv::AssignNodeOptionsIfAvailable(std::string* node_options) {
-  auto match = store_.find("NODE_OPTIONS");
+    // Default undefined or null to an empty string
+    std::string value = match[2].str();
 
-  if (match != store_.end()) {
-    *node_options = match->second;
-  }
-}
+    // Remove whitespace
+    value = std::regex_replace(value, std::regex("^\\s+|\\s+$"), "");
```

Comment:
1. Does this only remove whitespaces? If yes, can we avoid having regexes, and just delete using erase and remove. If not, can we make the comment a little bit more clear? What's the stop condition of this regex for example. Please document as much as you can to avoid any maintenance problems.
2. Can you define these regexes outside of the while loop?